src/snyk/common/configuration/configuration.ts

```diff
@@ -454,6 +458,15 @@ export class Configuration implements IConfiguration {
     );
   }
 
+  get scanningMode(): string {
+    const scanningModeInConfig = this.workspace.getConfiguration<string>(
+      CONFIGURATION_IDENTIFIER,
+      this.getConfigName(SCANNING_MODE),
+    );
+
+    return scanningModeInConfig ?? this.defaultScanningMode;
```

Comment:
Don't need to add null value handling here. This adds to the presentational layer logic that it shouldn't be responsible for. E.g. if LS decide to switch values of the setting, we'd have to go and touch this codebase.

1. You specify default in `package.json`. VS Code is responsible to provide it, if none is specified by a user.
2. It's LS's responsibility to handle whatever is defined in user `settings.json`. We should just be a proxy here.

Reply:
The problem is that `get scanningMode` returns a `string`, and the API call to fetch the setting can return undefined. So I have 3 options:
1. Return a `string | undefined` and handle the undefined case outside of the getter
2. Ensure it returns a string by handling the undefined case
3. Throw an exception when the return value is undefined

wdyt?

Reply:
I suggest option 1, and pass it down to LS. You define the type on LS settings as `string | undefined` already anyway.